### PR TITLE
doc(assump) : write docstrings for satask.py

### PR DESCRIPTION
**Pull request by JSS95**
#### Brief description of what is fixed or changed

Docstrings are made for the functions in `satask` module.
Some functions are split in order to make more readable code.
Some minor bugs related to default arguments of the functions are found and fixed.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Review by Soumi7 — COMMENTED** [review]
I don't think you should reorder the functions while adding a docstring. Also, have u built these changes to see how they look?

**Review by Soumi7 — COMMENTED** [review]
You shouldn't be changing the code. Just add docstrings where required.

File: sympy/assumptions/satask.py [hunk 1]
```diff
@@ -209,12 +208,47 @@ def find_symbols(pred):
     return pred.atoms(Symbol)
 
 
-def get_all_relevant_facts(proposition, assumptions=True,
-    context=global_assumptions, use_known_facts=True, iterations=oo):
+def get_all_relevant_facts(proposition, assumptions=None,
```
Comment (Soumi7) [inline sympy/assumptions/satask.py:265]: Please do not change the function definitions.

**Comment by JSS95** [comment]
@Soumi7 Codes in `satask.py` need some enhancement. I will open a separate PR for code change.

**Comment by Soumi7** [comment]
Yes please open a separate PR for code changes.

File: sympy/assumptions/satask.py [hunk 2]
```diff
@@ -9,6 +13,54 @@
...
+    will be found and passed to SAT solver because ``Q.nonnegative`` is
+    registered as a fact for ``Abs``.
+
+    Proposition is evaluated to ``True`` or ``False`` if the truth value can be
+    determined. If not, ``None`` is returned.
+
+    Parameters
+    ==========
+
+    proposition : any boolean expression
```
Comment (Soumi7) [inline sympy/assumptions/satask.py:31]: Change it to :

proposition : Any boolean expression.

File: sympy/assumptions/satask.py [hunk 3]
```diff
@@ -9,6 +13,54 @@
...
+        Proposition which will be evaluated to boolean value
```
Comment (Soumi7) [inline sympy/assumptions/satask.py:32]: Capitalize first letter and add fullstops for all.

**Review by Soumi7 — COMMENTED** [review]
Rest looks good to me

**Comment by JSS95** [comment]
Capitalized first letter and add fullstops. I'll enable auto-merge.

**Comment by asmeurer** [comment]
It's good to have things documented, but it should be noted that this is all internal API, and and could change at any point. satask is experimental, and its design shouldn't be considered solidified if we come up with better ideas. And even things like the `iterations` flag could end up being replaced with something better in the future.

**Pull request merged**